File: goblinframework-registry/src/main/java/org/goblinframework/registry/module/monitor/ZKP.java
package org.goblinframework.registry.module.monitor;

import org.goblinframework.core.monitor.AbstractInstruction;
import org.jetbrains.annotations.Nullable;

final public class ZKP extends AbstractInstruction {

  @Nullable public String operation;
  @Nullable public String path;
  @Nullable public String createMode;

  public ZKP() {
    super(Id.ZKP, Mode.SYN, true);
  }
}

File: goblinframework-registry-zookeeper/src/main/java/org/goblinframework/registry/zookeeper/module/monitor/ZKP.kt
package org.goblinframework.registry.zookeeper.module.monitor

import org.goblinframework.core.monitor.Instruction
import org.goblinframework.core.monitor.InstructionTranslator
import org.goblinframework.core.monitor.AbstractInstruction

class ZKP : AbstractInstruction(Instruction.Id.ZKP, Instruction.Mode.SYN, true) {

  var name: String? = null
  var operation: String? = null
  var path: String? = null
  var createMode: String? = null

  override fun translator(): InstructionTranslator {
    return InstructionTranslator {
      if (!it) {
        return@InstructionTranslator asShortText()
      }
      path?.run {
        val p = this
        createMode?.run {
          String.format("%s %s %s(%s) %s",
              asLongText(), name, operation, createMode, p)
        } ?: kotlin.run {
          String.format("%s %s %s %s",
              asLongText(), name, operation, p)
        }
      } ?: kotlin.run {
        String.format("%s %s %s",
            asLongText(), name, operation)
      }
    }
  }
}
